From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Wed, 16 May 2018 02:11:45 +0200
Subject: [PATCH] don't remove (tap|veth|fwpr) interfaces from bridge on ifup
 bridge

as proxmox don't defined them in /etc/network/interfaces
---
 ifupdown2/addons/bridge.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ifupdown2/addons/bridge.py b/ifupdown2/addons/bridge.py
index 1b4dcdd..3f97898 100644
--- a/ifupdown2/addons/bridge.py
+++ b/ifupdown2/addons/bridge.py
@@ -958,7 +958,7 @@ class bridge(moduleBase):
             runningbridgeports = self.brctlcmd.get_bridge_ports(ifaceobj.name)
             if runningbridgeports:
                 for bport in runningbridgeports:
-                    if not bridgeports or bport not in bridgeports:
+                    if ( not bridgeports or bport not in bridgeports ) and not re.match('^(tap|veth|fwpr)',bport):
                         self.ipcmd.link_set(bport, 'nomaster')
                         # set admin DOWN on all removed ports
                         # that don't have config outside bridge
-- 
2.11.0

